scripts/translation-status.ts

```diff
@@ -0,0 +1,437 @@
+		lines.push(['Page', ...this.targetLanguages].map((col) => `<th>${col}</th>`).join(''));
+		lines.push('</tr></thead>');
+
+		lines.push('<tbody>');
+		const spacer = `<tr class="spacer">\n${this.targetLanguages
+			.map(() => `<td></td>`)
+			.join('\n')}\n</tr>`;
+		lines.push(spacer);
+		statusByPage.forEach((content) => {
+			const cols = [];
+			cols.push(this.renderLink(content.githubUrl, content.subpath));
+			cols.push(
+				...this.targetLanguages.map((lang) => {
+					const translation = content.translations[lang];
+					if (translation.isMissing) return `<span title="${lang}: Missing">❌</span>`;
+					if (translation.isOutdated)
+						return `<a href="${translation.githubUrl}" title="${lang}: Needs updating">🔄</a>`;
+					return `<a href="${translation.githubUrl}" title="${lang}: Completed">✔</a>`;
+				})
+			);
+			lines.push(`<tr>\n${cols.map((col) => `<td>${col}</td>`).join('\n')}\n</tr>`);
+		});
+		lines.push(spacer);
+		lines.push('</tbody>');
+
+		lines.push('</table>');
+
+		lines.push(`\n<sup>❌ Missing &nbsp; 🔄 Needs updating &nbsp; ✔ Completed</sup>`);
+
+		return lines.join('\n');
+	}
+
+	/**
+	 * Render a link to a pre-filled GitHub UI for creating a new file.
+	 *
+	 * @param lang Language tag to create page for
+	 * @param filename Subpath of page to create
+	 */
+	renderCreatePageButton(lang: string, filename: string): string {
+		// We include `lang` twice because GitHub eats the last path segment when setting filename.
+		const createUrl = new URL(
+			`https://github.com/${this.githubRepo}/new/main/src/content/docs/${lang}`
+		);
+		createUrl.searchParams.set('filename', lang + '/' + filename);
+		createUrl.searchParams.set('value', '---\ntitle:\ndescription:\n---\n');
+		return `<strong><code>${this.renderLink(
+			createUrl.href,
+			`Create\xa0page\xa0+`
+		)}</code></strong>`;
+	}
+
+	/**
+	 * Render a progress bar with emoji.
+	 */
+	renderProgressBar(
+		total: number,
+		outdated: number,
+		missing: number,
+		{ size = 20 }: { size?: number } = {}
+	) {
+		const outdatedLength = Math.round((outdated / total) * size);
+		const missingLength = Math.round((missing / total) * size);
+		const doneLength = size - outdatedLength - missingLength;
+		return [
+			[doneLength, '🟪'],
+			[outdatedLength, '🟧'],
+			[missingLength, '⬜'],
+		]
+			.map(([length, icon]) => Array(length).fill(icon))
+			.flat()
+			.join('');
+	}
+
+	renderLink(href: string, text: string): string {
+		return `<a href="${escape(href)}">${escape(text)}</a>`;
+	}
+}
+
+const translationStatusBuilder = new TranslationStatusBuilder({
+	pageSourceDir: './src/content/docs',
+	htmlOutputFilePath: './dist/translation-status/index.html',
+	sourceLanguage: 'en',
+	targetLanguages: ['ar', 'de', 'es', 'fr', 'ja', 'ko', 'pl', 'pt-br', 'zh-cn'],
+	languageLabels: {
+		ar: 'العربية',
+		de: 'Deutsch',
+		es: 'Español',
+		fr: 'Français',
+		ja: '日本語',
+		ko: '한국어',
+		pl: 'Polski',
+		'pt-br': 'Português do Brasil',
+		'zh-cn': '简体中文',
+	},
```

Comment:
While we’re here can we import these from `src/i18n/languages.ts`? We recently discussed that there’s no particular reason to exclude some languages from that file so should simplify maintenance a bit.

```suggestion
	targetLanguages: Object.keys(languages),
	languageLabels: languages,
```

Reply:
Great suggestion, I added that with some minor tweaks!